sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,1020 @@
+    if (response == AZ_ERROR_ITEM_NOT_FOUND)
+    {
+      return_code = 404;
+    }
+    else
+    {
+      return_code = 200;
+    }
+
+    // Send command response with report as JSON payload
+    int rc;
+    if ((rc = send_command_response(command_request, return_code, command_response_span)) != 0)
+    {
+      printf("Unable to send %u response, status %d\n", return_code, rc);
+    }
+  }
+  else
+  {
+    // Unsupported command
+    printf(
+        "Unsupported command received: %.*s.\n",
+        az_span_size(command_request->name),
+        az_span_ptr(command_request->name));
+
+    int rc;
+    if ((rc = send_command_response(command_request, 501, report_error_payload)) != 0)
+    {
+      printf("Unable to send %d response, status %d\n", 501, rc);
+    }
+  }
+}
+
+// Callback for incoming MQTT messages
+static int on_received(char* topicName, int topicLen, MQTTClient_message* message)
+{
+  if (topicLen == 0)
+  {
+    // The length of the topic if there are one or more NULL characters embedded in topicName,
+    // otherwise topicLen is 0.
+    topicLen = (int)strlen(topicName);
+  }
+
+  printf("Topic: %s\n", topicName);
+
+  az_span topic_span = az_span_init((uint8_t*)topicName, topicLen);
+
+  // Parse the incoming message topic and check which feature it is for
+  az_iot_hub_client_twin_response twin_response;
+  az_iot_hub_client_method_request command_request;
+
+  if (az_succeeded(
+          az_iot_hub_client_twin_parse_received_topic(&client, topic_span, &twin_response)))
+  {
+    printf("Twin Message Arrived\n");
+
+    // Determine what kind of twin message it is and take appropriate actions
+    handle_twin_message(message, &twin_response);
+
+    printf("Response status was %d\n", twin_response.status);
+  }
+  else if (az_succeeded(az_iot_hub_client_methods_parse_received_topic(
+               &client, az_span_init((uint8_t*)topicName, topicLen), &command_request)))
+  {
+    printf("Command arrived\n");
+
+    // Determine if the command is supported and take appropriate actions
+    handle_command_message(message, &command_request);
+  }
+
+  putchar('\n');
+
+  return 1;
+}
+
+static int connect_device(void)
+{
+  int rc;
+
+  MQTTClient_SSLOptions mqtt_ssl_options = MQTTClient_SSLOptions_initializer;
+  MQTTClient_connectOptions mqtt_connect_options = MQTTClient_connectOptions_initializer;
+
+  // NOTE: We recommend setting clean session to false in order to receive any pending messages
+  mqtt_connect_options.cleansession = false;
+  mqtt_connect_options.keepAliveInterval = AZ_IOT_DEFAULT_MQTT_CONNECT_KEEPALIVE_SECONDS;
+
+  // Get the MQTT username used to connect to IoT Hub
+  if (az_failed(
+          rc = az_iot_hub_client_get_user_name_with_model_id(
+              &client, model_id, mqtt_username, sizeof(mqtt_username), NULL)))
+
+  {
+    printf("Failed to get MQTT username, return code %d\n", rc);
+    return rc;
+  }
+
+  printf("%s\r\n", mqtt_username);
```

Comment:
>printf [](start = 2, length = 6)

Nit: We should tell user what we're outputing, so like printf("MQTT username: %s\n",...)

Reply:
true